### PR TITLE
[Mobile] Feat/Auth

### PR DESCRIPTION
**Pull request by henrique-mariano**
Add auth screens on mobile

### CODE REVIEW

File: mobile/src/screens/auth/sign-in/index.tsx [hunk 1]
```diff
@@ -1,5 +1,71 @@
-import { View } from "react-native";
+import { useNavigation } from "@react-navigation/native";
+import { useState } from "react";
+import { StyleSheet, View } from "react-native";
+import { Button, Input } from "react-native-elements";
```
Comment (yuriserka) [inline mobile/src/screens/auth/sign-in/index.tsx:4]: We're going to use some UI library? Have you searched others?

File: mobile/src/store/supabase/index.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { createClient } from "@supabase/supabase-js";
+
+// TODO: sensitive data must be in .env
+const supabaseUrl = "https://ztznlnawnncaxlmnhhjm.supabase.co/";
+const supabaseAnonKey = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Inp0em5sbmF3bm5jYXhsbW5oaGptIiwicm9sZSI6ImFub24iLCJpYXQiOjE2NTU3NjYzMjQsImV4cCI6MTk3MTM0MjMyNH0.GGNH3MyOf_4n2V1r_LR0kqrH-YjYi0HK9oHHMzujGxI";
```
Comment (yuriserka) [inline mobile/src/store/supabase/index.ts:6]: this should be in a `.env` already

File: mobile/src/store/auth/reducer.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { UserAuthState } from "./user";
 
 export interface UserAuthAction {
-  type: "LOGIN" | "ERROR";
+  type: "LOGIN" | "ERROR" | "SIGNIN" | "SIGNUP";
   payload?: Partial<UserAuthState>;
   error?: string;
 }
```
Comment (yuriserka) [inline mobile/src/store/auth/reducer.ts:7]: may we need to refactor this to something more generic or redux like

File: mobile/src/store/location/reducer.ts [hunk 4]
```diff
@@ -1,7 +1,12 @@
...
-  type: "UPDATE_POS" | "UPDATE_ZOOM" | "FIND_ME" | "ERROR";
+  type:
+    | "UPDATE_POS"
+    | "UPDATE_ZOOM"
+    | "FIND_ME"
+    | "ERROR"
+    | "UPDATE_USER_MARKER_INFO";
   payload?: Partial<UserLocationState>;
   error?: string;
 }
```
Comment (yuriserka) [inline mobile/src/store/location/reducer.ts:12]: review

**Review by yuriserka — CHANGES_REQUESTED** [review]

File: mobile/src/screens/auth/sign-in/index.tsx (hunk 1 above)
Comment (henrique-mariano) [reply mobile/src/screens/auth/sign-in/index.tsx:4]: We gonna usehttps://nativebase.io

File: mobile/src/store/auth/provider.tsx [hunk 5]
```diff
@@ -17,8 +16,8 @@ export interface UserAuthContextProps {
 const UserAuthContext = createContext<UserAuthContextProps>(undefined!);
 
 const userAuthInitialState: UserAuthState = {
-  user: null,
-  error: null,
+  user: undefined,
+  error: undefined,
 };
 
 export default function UserAuthProvider({ children }: PropsWithChildren<{}>) {
```
Comment (henrique-mariano) [inline mobile/src/store/auth/provider.tsx:23]: Move action to provider

**Pull request merged**